plugins/woocommerce/includes/admin/class-wc-admin-marketplace-promotions.php

```diff
@@ -0,0 +1,266 @@
+<?php
+/**
+ * Addons Page
+ *
+ * @package  WooCommerce\Admin
+ * @version  2.5.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * WC_Admin_Marketplace_Promotions class.
+ */
+class WC_Admin_Marketplace_Promotions {
+
+	const TRANSIENT_NAME = 'wc_addons_marketplace_promotions';
+	public static string $locale = 'fr_FR';
+
+	/**
+	 * On selected WooCommerce admin pages, fetch promotions from transient or API.
+	 * On pages in the main WooCommerce menu, show promotions if there are any.
+	 *
+	 * @return void
+	 */
+	public static function init_marketplace_promotions() {
+		if ( self::is_targeted_page_to_fetch_promotions() ) {
+			self::fetch_marketplace_promotions();
+		}
+
+		if ( self::is_targeted_page_to_show_promotions() ) {
+			self::$locale = self::$locale ?? get_user_locale();
+			if ( empty( self::$locale ) ) {
+				self::$locale = 'en_US';
+			}
+			self::show_promotions();
+		}
+	}
+
+	/**
+	 * Check if the request is for one of the pages we will run fetch_marketplace_promotions
+	 * on: the WooCommerce Home and Extensions pages.
+	 *
+	 * @return bool
+	 */
+	public static function is_targeted_page_to_fetch_promotions(): bool {
+		if ( defined( 'DOING_AJAX' ) && DOING_AJAX ) {
+			return false;
+		}
+
+		if ( ! is_admin() ) {
+			return false;
+		}
+
+		if ( ! isset( $_GET['page'] ) ) {
+			return false;
+		}
+
+		if ( 'wc-admin' === $_GET['page'] ) {
+			if ( ! isset( $_GET['path'] ) ) {
+				return true;
+			}
+			if ( '/extensions' === $_GET['path'] ) {
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Check if the request is for one of the pages we will show promotions
+	 * on: the pages in the main WooCommerce menu, excluding Analytics and Marketing.
+	 *
+	 * @return bool
+	 */
+	public static function is_targeted_page_to_show_promotions(): bool {
+		if ( defined( 'DOING_AJAX' ) && DOING_AJAX ) {
+			return false;
+		}
+
+		if ( ! is_admin() ) {
+			return false;
+		}
+
+		if ( ! isset( $_GET['page'] ) ) {
+			return false;
+		}
```

Comment:
Thos could be merged.

Reply:
Thanks, done in c8fc24668b79740d10263a092edef91da68c3dbf.